File: src/main/kotlin/dev/loveeev/astraTowny/timers/NewHourEvent.java
package dev.loveeev.astraTowny.timers;

import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

public class NewHourEvent extends Event {
    private static final HandlerList handlers = new HandlerList();
    private final int hour;

    public NewHourEvent(int year,int month,String day,int hour) {
        this.hour = hour;
    }

    public int getHour() {
        return hour;
    }

    public HandlerList getHandlers() {
        return handlers;
    }

    public static HandlerList getHandlerList() {
        return handlers;
    }
}

File: src/main/kotlin/dev/loveeev/astraTowny/timers/NewHourEvent.kt
package dev.loveeev.astratowny.timers

import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class NewHourEvent(year: Int, month: Int, day: String?, val hour: Int) : Event() {

    override fun getHandlers(): HandlerList {
        return handlerList
    }

    companion object {
        val handlerList: HandlerList = HandlerList()
    }
}